yt/visualization/fixed_resolution_filters.py

```diff
@@ -21,7 +21,12 @@
 def apply_filter(f):
     @wraps(f)
     def newfunc(*args, **kwargs):
-        args[0]._filters.append((f.__name__, (args, kwargs)))
+        frb = args[0]
+        frb._filters.append((f.__name__, (args, kwargs)))
+        # We need to invalidate the data to force the plot window to
+        # take the filter into account.
+        if frb.plot_window:
+            frb.plot_window._data_valid = False
```

Comment:
What if FixedResolutionBuffer instances had a `_data_valid` attribute. We would need to refactor PlotWindow to store that state on the `frb` attribute but I think it would be a straightforward refactor.